File: FreakFightsFan.Shared/Features/Fights/Responses/FightDto.cs
using FreakFightsFan.Shared.Features.DictionaryItems.Responses;
using FreakFightsFan.Shared.Features.Events.Responses;
using FreakFightsFan.Shared.Features.Teams.Responses;

namespace FreakFightsFan.Shared.Features.Fights.Responses;

public class FightDto
{
    public int Id { get; init; }
    public DateTime Created { get; set; }
    public DateTime Modified { get; set; }
    public int OrderNumber { get; set; }
    public string VideoUrl { get; set; }

    public int EventId { get; set; }
    public EventDto Event { get; set; }
    public List<TeamDto> Teams { get; set; }
    public MyDictionaryItemDto? Type { get; set; }
}

File: freak-fights-fan/shared/features/fights/responses/FightDto.ts
import { MyDictionaryItemDto } from "../../dictionaryItems/responses/MyDictionaryItemDto";
import { EventDto } from "../../events/responses/EventDto";
import { TeamDto } from "../../teams/responses/TeamDto";

export interface FightDto {
    id: number;
    created: Date;
    modified: Date;
    orderNumber: number;
    videoUrl: string;

    eventId: number;
    event: EventDto;
    teams: Array<TeamDto>;
    type: MyDictionaryItemDto | undefined;
}